rtl/eclrtl/rtlfield.cpp

```diff
@@ -339,12 +384,23 @@ size32_t RtlIntTypeInfo::build(ARowBuilder &builder, size32_t offset, const RtlF
 
 size32_t RtlIntTypeInfo::buildInt(ARowBuilder &builder, size32_t offset, const RtlFieldInfo *field, __int64 val) const
 {
-    builder.ensureCapacity(length+offset, field->name);
+    builder.ensureCapacity(length+offset, queryName(field));
     rtlWriteInt(builder.getSelf() + offset, val, length);
     offset += length;
     return offset;
 }
 
+size32_t RtlIntTypeInfo::buildString(ARowBuilder &builder, size32_t offset, const RtlFieldInfo *field, size32_t size, const char *value) const
+{
+    return buildInt(builder, offset, field, rtlStrToInt8(size, value));
```

Comment:
Should this take any account of the signedness of the target field?

Reply:
I don't think it needs to string to int should work for (-2^(n-1)..2^n-1